src/Fixer/Whitespace/ArrayIndentationFixer.php

```diff
@@ -0,0 +1,331 @@
+    {
+        return new FixerDefinition(
+            'Multiline arrays must be indented with configured indentation.',
+            [
+                new CodeSample("<?php\n\$foo = [\n   'bar' => [\n    'baz' => true,\n  ],\n];\n"),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isAnyTokenKindsFound([T_ARRAY, CT::T_ARRAY_SQUARE_BRACE_OPEN]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPriority()
+    {
+        // should run after BracesFixer
+        return -30;
+    }
+
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        foreach ($this->findArrays($tokens) as $array) {
+            $previousLineBracesDelta = 1;
+            $indentLevel = 0;
+            $ignoredPositiveDelta = 0;
+            $arrayIndent = $this->getLineIndentation($tokens, $array['start']);
+            $previousLineInitialIndent = $arrayIndent;
+            $previousLineNewIndent = $arrayIndent;
+
+            foreach ($array['line_braces_deltas'] as $index => $lineBracesDelta) {
+                $token = $tokens[$index];
+
+                $extraIndentLevel = 0;
+
+                if ($previousLineBracesDelta > 0) {
+                    ++$indentLevel;
+
+                    if ($previousLineBracesDelta > 1) {
+                        $ignoredPositiveDelta += $previousLineBracesDelta - 1;
+                    }
+                }
+
+                if ($lineBracesDelta < 0) {
+                    $delta = $lineBracesDelta + $ignoredPositiveDelta;
+                    $ignoredPositiveDelta = max(0, $ignoredPositiveDelta + $lineBracesDelta);
+                    $indentLevel = max(0, $indentLevel + $delta);
+
+                    if ($this->isClosingLineWithMeaningfulContent($tokens, $index)) {
+                        $extraIndentLevel = -$lineBracesDelta;
+                    }
+                }
+
+                if ($this->newlineIsInArrayScope($tokens, $index, $array)) {
+                    $content = preg_replace(
+                        '/(\R)[\t ]*$/',
+                        '$1'.$arrayIndent.str_repeat($this->whitespacesConfig->getIndent(), $indentLevel + $extraIndentLevel),
+                        $token->getContent()
+                    );
+
+                    $previousLineInitialIndent = $this->extractIndent($token->getContent());
+                    $previousLineNewIndent = $this->extractIndent($content);
+                } else {
+                    $content = preg_replace(
+                        '/(\R)'.preg_quote($previousLineInitialIndent, '/').'([\t ]*)$/',
+                        '$1'.$previousLineNewIndent.'$2',
+                        $token->getContent()
+                    );
+                }
+
+                $previousLineBracesDelta = $lineBracesDelta;
+
+                $tokens[$index] = new Token([T_WHITESPACE, $content]);
```

Comment:
just double-checking, does this fixer any time inserting new token to the collection?

Reply:
It doesn't.